utils.py

```diff
@@ -126,11 +126,11 @@ def get_service_ports(self, app):
                 if ports:
                     return list(ports)
 
-        ports = app.get('ports', [])
```

Comment:
Instead of these, I would instead remove all None values from the initial JSON.

So something like:
```
Python 3.6.2 (default, Jul 17 2017, 16:44:45)
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import json
>>> json_value = '{"key":null,"key2":"yyy"}'
>>> def remove_nulls(d):
...      return {k: v for k, v in d.items() if v is not None}
...
>>> json.loads(json_value, object_hook=remove_nulls)
{'key2': 'yyy'}
```

And if we do this enough it may be worth it to have a wrapper for loading JSON